test/e2e/toolchaincluster_test.go

```diff
@@ -107,9 +130,8 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 		// given
 		name := generateNewName("new-offline-", current.Name)
 		secretCopy := &corev1.Secret{}
-		wait.CopyWithCleanup(t, await,
-			client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace},
-			client.ObjectKey{Name: name, Namespace: current.Namespace}, secretCopy)
+		wait.CopyWithCleanup(t, await, client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace},
+			client.ObjectKey{Name: name, Namespace: current.Namespace}, secretCopy, kubeconfig.Modify(t, kubeconfig.ApiEndpoint("https://1.2.3.4:8443")))
```

Comment:
minor thing - unless I've missed something the `ApiEndpoint` from the `secretCopy` kubeconfig is not actually used.

Reply:
Au contraire, the `secretCopy` is stored with a modified kubeconfig that contains an invalid api endpoint. The test then makes sure that the toolchain cluster CR that reference that secret correctly reports the status as not ready, because the api endpoint (obtained from the kubeconfig) is invalid.

Reply:
I got confused since also the toolchaincluster resource gets created with the same API endpoint, my bad. Thanks for clarifying that.

Maybe a small comment explaining why are we setting that api endpoint would help , something like: _"set invalid api endpoint so that tc goes into unreachable state"_ or something like that.